### ISSUE TITLE
multi gpu 분산 학습구조 개선

### ISSUE THREAD

**Issue by paranoidandroid2124** [Intelligent-Agent-Lab/ARC_PRIZE_2025#73]
크게 요약: 
1. 멀티 GPU로 forward만 분산하고, gradient 집계 및 업데이트는 cuda:0 한 곳에서만 수행
2. 다른 GPU에서 계산된 grad는 업데이트에 반영되지 않고, 롤아웃 또한 cuda:0만 사용
3. gradient accumulation : zero_grad 호출 위치가 문제 있음

-> 각 GPU가 자신의 미니배치로 forward/backward 모두 수행하고 grad 값 평균 낸 뒤에 업데이트 + 롤아웃 병렬로 수행 구현 필요

코드 부분:
1. 롤아웃은 cuda 0에서 하고있음.
```
# setup_agent()
self.device = torch.device("cuda:0")
self.agent = Agent(self.config).to(self.device)

# train() → collect_rollouts()
next_obs = torch.Tensor(next_obs).to(self.device)  # cuda:0
with torch.no_grad():
    action, logprob, _, value = self.agent.get_action_and_value(next_obs, obs_info=current_infos)  # cuda:0
```

2. 업데이트 단계의 멀티 GPU forward 후, 결과를 cuda:0로 모아 cat
```
# process_batch_multi_gpu()
chunk_result = agent_to_use.get_action_and_value(chunk_obs, chunk_actions)
chunk_result = [t.to(self.device) if t is not None else None for t in chunk_result]
...
# Concatenate results
concatenated.append(torch.cat(tensors, dim=0))   # cuda:0에서 cat
# loss.backward() 시 각 경로의 grad는 각 레플리카 파라미터로 
```

3. optimizer는 cuda 0 모델만 관리
```
# setup_agent()
self.optimizer = optim.Adam(
    self.agent.parameters(),   # self.agent = cuda:0
    lr=self.config.training.learning_rate,
    eps=1e-5
)
# → cuda:0 파라미터만 step/zero 대상. 다른 GPU grad는 집계·반영되지 않음.
```
